packages/web/src/pages/upload-page/pages/SelectPage.tsx

```diff
@@ -42,8 +42,10 @@ export const SelectPageNew = (props: SelectPageProps) => {
   const onSelectTracks = useCallback(
     async (selectedFiles: File[]) => {
       const existing = new Set(
-        tracks.map(
-          ({ file }: { file: File }) => `${file.name}-${file.lastModified}`
+        tracks.map(({ file }: { file: File | NativeFile }) =>
+          'lastModified' in file
+            ? `${file.name}-${file.lastModified}`
+            : file.name
```

Comment:
we could add a hash or something here to prevent dupes, or track recurring names and increment them

I found out you can actually provide 2+ matching file names if they come from different paths

Reply:
gonna cast this instead - lastModified is always defined on web